avalancheup-aws/src/apply/mod.rs

```diff
@@ -1440,23 +1440,31 @@ pub async fn execute(log_level: &str, spec_file_path: &str, skip_prompt: bool) -
                 .unwrap();
 
             log::info!("waiting for anchor nodes bootstrap and ready (to be safe)");
-            sleep(Duration::from_secs(15)).await;
-
-            for ssh_command in ssh_commands.iter() {
-                match ssh_command.run("tail -10 /var/log/cloud-init-output.log") {
-                    Ok(output) => {
-                        println!(
-                            "{} (anchor node) init script std output:\n{}\n",
-                            ssh_command.instance_id, output.stdout
-                        );
-                        println!(
-                            "{} (anchor node) init script std err:\n{}\n",
-                            ssh_command.instance_id, output.stderr
-                        );
+            sleep(Duration::from_secs(15)).await;          
+            async fn read_script(ssh_commands: Vec<ec2::SshCommand>) {
+                for ssh_command in ssh_commands.iter() {
+                    match ssh_command.run("tail -10 /var/log/cloud-init-output.log") {
+                        Ok(output) => {
+                            println!(
+                                "{} (anchor node) init script std output:\n{}\n",
+                                ssh_command.instance_id, output.stdout
+                            );
+                            println!(
+                                "{} (anchor node) init script std err:\n{}\n",
+                                ssh_command.instance_id, output.stderr
+                            );
+                        }
+                        Err(e) => log::warn!("failed to run ssh command {}", e),
                     }
-                    Err(e) => log::warn!("failed to run ssh command {}", e),
                 }
             }
+            let timeout_duration = Duration::from_secs(10);
+            let result = tokio::time::timeout(timeout_duration, read_script(ssh_commands)).await;
+            if result.is_err() {
```

Comment:
This is ok but really Go style versus Rust. I would recommend using `?` or `.map_err()` for handling the error case.

Reply:
`?` o `map_err` will return here? The point here is to ignore the error

Reply:
hmm what about `.map_err(|_| log::warn!(...)`

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=43a8ac893765dca591ea8e717bf59831